packages/cozy-client/src/models/konnectorFolder.js

```diff
@@ -0,0 +1,213 @@
+import get from 'lodash/get'
+import trim from 'lodash/trim'
+
+import { folder as folderModel, account as accountModel } from '.'
+
+const FILES_DOCTYPE = 'io.cozy.files'
+const PERMISSIONS_DOCTYPE = 'io.cozy.permissions'
+
+export const ensureKonnectorFolder = async (client, { konnector, account }) => {
+  const permissions = client.collection(PERMISSIONS_DOCTYPE)
+  const files = client.collection(FILES_DOCTYPE)
+  const adminFolder = await folderModel.ensureMagicFolder(
+    client,
+    folderModel.MAGIC_FOLDERS.ADMINISTRATIVE,
+    '/Administratif'
+  )
+  const path = buildFolderPath(konnector, account, {
+    // @ts-ignore
+    administrative: adminFolder.path
+  })
+  const folder =
+    (await statDirectoryByPath(client, path)) ||
+    (await createDirectoryByPath(client, path))
+
+  await permissions.add(konnector, buildFolderPermission(folder))
+  await files.addReferencesTo(konnector, [folder])
+
+  return folder
+}
+
+/**
+ * Creates a directory from a given path
+ *
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @returns {Promise<import('../types').FileDocument>}         Directory attributes
+ */
+export const createDirectoryByPath = async (client, path) => {
+  const { data } = await client
+    .collection(FILES_DOCTYPE)
+    .createDirectoryByPath(path)
+  return data
+}
+
+/**
+ * Retrieves a directory from its path
+ *
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @returns {Promise<import('../types').FileDocument>}        Created io.cozy.files document
+ */
+export const statDirectoryByPath = async (client, path) => {
+  try {
+    const response = await client.collection(FILES_DOCTYPE).statByPath(path)
+    return response.data
+  } catch (error) {
+    if (error && error.status === 404) return null
+    throw new Error(error.message)
+  }
+}
+
+/**
+ * Build folder path for a given konnector and a given account.
+ *
+ * If konnector.folders[0].defaultDir exists, it is used as default directory.
+ *
+ * Occurrences of following strings in base directory are replaced by:
+ * * `$administrative`: Administrative folder
+ * * `$photos`: Photos folder
+ *
+ * Occurrences of following strings in path are replaced by:
+ * * `$account: Account label (id or name)`
+ * * `$konnector`: Konnector name
+ *
+ * If no konnectors.folders[0].defaultDir is set, the default dir used is
+ * *  `$administrative/$konnector/$account`
+ *
+ * @param  {Object} konnector Konnector document
```

Comment:
IoCozyKonnector / IoCozyAccount.

You can also create a type for "folders" here.

Reply:
I don't know how to create a type for this "folders" object, since it can have any key with any localized name. I renamed it to "magicFolders" to make it clearer I hope.

Reply:
Can you give me an exemple of the content? 

Because if it's just : 
```
{
'fr': 'foo',
'en': 'bar
}
```

then `Object<string, string>` should be OK. 

But it seems that we can read value of `folders[0].defaultDir` so I'm not sure about the structure...

Reply:
Related to this, I found out that the "localization" feature for magic folders was missing. Now it is done.
Now the jsdoc is also updated with the possible values in the object. Which was not so complicated after all.